keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -259,66 +248,86 @@ the API:
-kind: Gateway
+kind: ClusterReferenceGrant
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
-  name: cross-namespace-tls-gateway
-  namespace: gateway-api-example-ns1
-spec:
-  gatewayClassName: acme-lb
-  listeners:
-  - name: https
-    protocol: HTTPS
-    port: 443
-    hostname: "*.example.com"
-    tls:
-      # There's a Kind/Resource mismatch here, which sucks, but it is not
-      # easily fixable, since Gateway is already a beta, close to GA
-      # object.
-      certificateRefs:
-      - kind: Secret
-        group: ""
-        name: wildcard-example-com-cert
-        namespace: gateway-api-example-ns2
----
-apiVersion: authorization.k8s.io/v1alpha1
+  name: gateway-tls
+from:
+  group: gateway.networking.k8s.io
+  resource: gateways
+  version: v1
+  path: ".spec.listeners[*].tls.certificateRefs[*]"
+to:
+  group: ""
+  resource: secrets
+for: tls
+```
+
+Implementations of APIs can bundle ClusterReferenceConsumer resources as part of
+their deployment. This resource links a subject (likely the ServiceAccount
+used by the controller) to matching (Cluster)ReferenceGrants. The subject of
+this ClusterReferenceConsumer will receive authorization for any
+(Cluster)ReferenceGrant with matching `from`, `to`, and `for` values.
+
+```yaml
+kind: ClusterReferenceConsumer
+apiVersion: reference.authorization.k8s.io/v1alpha1
+metadata:
+  name: contour-gateway
+subject:
+  kind: ServiceAccount
+  name: contour
+  namespace: contour-system
+from:
+  group: gateway.networking.k8s.io
+  resource: gateways
+to:
+  group: ""
+  resource: secrets
+for: tls
+```
+
+Finally, users can use ReferenceGrant to authorize specific cross-namespace
+references. For example, the following ReferenceGrant authorizes references from
+Gateways in the `prod` namespace to the `acme-tls` Secret in the `prod-tls`
+namespace:
+
+```yaml
 kind: ReferenceGrant
+apiVersion: reference.authorization.k8s.io/v1alpha1
 metadata:
-  name: allow-ns1-gateways-to-ref-secrets
-  namespace: gateway-api-example-ns2
-spec:
-  from:
-  - group: gateway.networking.k8s.io
-    resource: gateways
-    namespace: gateway-api-example-ns1
-  to:
-  - group: ""
-    resource: secrets
+  name: prod-gateways
+from:
+  group: gateway.networking.k8s.io
+  resource: gateways
+  namespace: prod
+to:
+  group: ""
+  resource: secrets
+  names:
+  - acme-tls
+for: tls
 ```
 
-For Gateway TLS references, if this ReferenceGrant is deleted (revoking, 
-the grant), then the Listener will become invalid, and the configuration
-will be removed as soon as possible (eventual consistency permitting).
+### Existing ReferenceGrant Usage Examples
 
 #### Gateway API HTTPRoute Referencing Service
 
 In this example, a HTTPRoute in the `baz` namespace is directing traffic
 to a Service backend in the `quux` namespace.
 
 ```yaml
-apiVersion: gateway.networking.k8s.io/v1beta1
+apiVersion: gateway.networking.k8s.io/v1
 kind: HTTPRoute
 metadata:
   name: quuxapp
```

Comment:
```suggestion
  name: quux-route
```

to make it easier to read, now the route and the service have the same name which makes it hard to read.

Reply:
Updated, thanks!